src/traces/table/plot.js

```diff
@@ -0,0 +1,82 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var table = require('./render');
+
+module.exports = function plot(gd, cdTable) {
+
+    var fullLayout = gd._fullLayout;
+    var svg = fullLayout._paper;
+    var root = fullLayout._paperdiv;
+
+    var gdColumns = {};
+    var gdColumnsOriginalOrder = {};
+
+    var size = fullLayout._size;
+
+    cdTable.forEach(function(d, i) {
+        gdColumns[i] = gd.data[i].header.values.map(function(d) {return d[0];});
```

Comment:
With the `render.js` / `plot.js` merge, these calculation steps have moved to `calc.js`.

I'd need a bit of help though. I [persisted stuff](https://github.com/plotly/plotly.js/blob/table/src/traces/table/plot.js#L260) on `gd.calcdata` as you suggested but upon a subsequent `Plotly.restyle` (of some unrelated attribute) it'll be overwritten as though I fully re-stated the `table.json`, so the information is lost. Is there a very simple example for showing how to persist things so they don't get lost on unrelated attribute change?

Somewhat related: to expose column ordering to the user, and also to let the users easily read the column order, this new attribute `columnorder` had been added.

Unrelated question: is there a way for `plot.js` or `calc.js` to know _what_ changed? For example, if the user called `Plotly.restyle` with `line.color` or some such, then a lot of the logic and rendering may not need to be redone.